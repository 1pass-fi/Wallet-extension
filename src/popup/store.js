import { composeWithDevTools } from 'redux-devtools-extension'
import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'

import rootReducer from './reducers/index'

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
)

export default store
