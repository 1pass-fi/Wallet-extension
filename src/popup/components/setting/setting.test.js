import '@babel/polyfill'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { Provider } from 'react-redux'
import store from 'popup/store'

import Setting from '.'

describe('Test for Setting component', () => {
  describe('Render without crashing', () => {
    it('renders correctly', () => {
      const { container } = render(<Router><Provider store={store}><Setting /></Provider></Router>)
      expect(container).toMatchSnapshot()
    })
  })
})
