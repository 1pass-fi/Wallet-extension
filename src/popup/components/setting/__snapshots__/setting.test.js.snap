// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Setting component Render without crashing renders correctly 1`] = `
<div>
  <div
    class="setting-container"
  >
    <div>
      <button
        class="global-button"
      >
        <div
          class="global-button-content"
        >
          <div
            class="global-button-icon"
          >
            <svg />
          </div>
          <div
            class="global-button-label"
          >
            Lock Koi Wallet
          </div>
        </div>
      </button>
    </div>
    <div
      class="setting-mock-content"
    >
      <div
        class="account-setting-row"
      >
        <div
          class="display-row icon"
        >
          <div
            class="logo-icon"
          >
            <svg />
          </div>
        </div>
        <div
          class="display-row account-info"
        >
          <div
            class="account-name-line"
          >
            <div
              class="name"
            >
              Account 1
            </div>
            <div
              class="edit-icon"
            >
              <svg />
            </div>
          </div>
          <button
            class="button-shared button-outline get-seed-phrase-button"
          >
            Get Seed Phrase
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
